### ISSUE TITLE
Is it possible to set the precision of float parameters in a model?

### ISSUE THREAD

**Issue by onurgu** [clab/dynet#1225]
Is there a configuration parameter in DyNet to set the precision of float parameters, for example as in Theano's `floatX`?

**Comment by neubig** [comment]
No. This is functionality that we have wanted for some time, but it'd require significant refactoring. I'll mark it as an enhancement.

**Comment by onurgu** [comment]
Sorry to know that.

Can you suggest any workarounds for this? I suspect this might affect the convergence behavior for some models.

**Comment by neubig** [comment]
There is no workaround as far as I know, but I have not encountered a neural model where the DyNet default of 32 bit precision is a significant problem (if anything, it's common for people to use 16 bits to save memory).

**Comment by onurgu** [comment]
I thought that the default precision was 64 or 53 bits so that I could not replicate another work which uses 32 bit of precision in floats.

I tend to ask whether you have any suggestions for replicating another model in another framework, however I know that is uncharted area with lots of complex details.
